src/extension/prompt/node/chatMLFetcher.ts

```diff
@@ -136,9 +136,10 @@ export class ChatMLFetcherImpl extends AbstractChatMLFetcher {
 			ignoreStatefulMarker: opts.ignoreStatefulMarker
 		});
 		let tokenCount = -1;
+		const streamRecorder = new FetchStreamRecorder(finishedCb);
+		const enableRetryOnError = opts.enableRetryOnError || (opts.enableRetryOnFilter && opts.enableRetryOnError === undefined);
```

Comment:
And, isn't this much simpler?

```suggestion
		const enableRetryOnError = opts.enableRetryOnError ?? opts.enableRetryOnFilter;
```